container_pipeline/workers/dockerfile_lint.py

```diff
@@ -0,0 +1,127 @@
+#!/usr/bin/env python
+
+import json
+import logging
+import os
+
+from container_pipeline.lib.log import load_logger
+from container_pipeline.workers.base import BaseWorker
+from container_pipeline.lib import settings
+from container_pipeline.lib.command import run_cmd
+
+
+class DockerfileLintWorker(BaseWorker):
+    """
+    Dockerfile lint worker lints the Dockerfile using the projectatomic's
+    dockerfile_lint tool. https://github.com/projectatomic/dockerfile_lint
+    """
+
+    def handle_job(self, job):
+        """
+        This menthod handles the job received for dockerfile lint worker.
+        """
+        self.logger.info("Received job for Dockerfile lint.")
+        self.logger.debug("Job data: %s" % job)
+        self.logger.info("Writing Dockerfile to /tmp/scan/Dockerfile")
+        self.write_dockerfile(job.get("Dockerfile"))
+        self.logger.info("Running Dockerfile Lint check")
+        self.lint(job)
+
+    def write_dockerfile(self, dockerfile):
+        """
+        Write dockerfile at temporary location to run lint upon
+        """
+        if os.path.isdir("/tmp/scan"):
+            self.logger.info("/tmp/scan directory already exists")
+        elif os.path.isfile("/tmp/scan"):
+            os.remove("/tmp/scan")
+            os.makedirs("/tmp/scan")
+        else:
+            os.makedirs("/tmp/scan")
+
+        with open("/tmp/scan/Dockerfile", "w") as f:
+            f.write(dockerfile)
+
+    def lint(self, job):
+        """
+        Lint the Dockerfile received
+        """
+        command = ["docker", "run", "--rm", "-v", "/tmp/scan:/root/scan:Z",
+                   "registry.centos.org/pipeline-images/dockerfile-lint"]
+
+        try:
+            out = run_cmd(command)
+        except Exception as e:
+            self.logger.error(
+                "Dockerfile Lint check failed", extra={'locals': locals()})
```

Comment:
I think yes, its all about logging verbose debug information at the time of exception.